### PR TITLE
refactor(kubernetes): namespace selector component

### PR DESCRIPTION
**Pull request by ethanfrogers**
create a namespace selector component which can be reused. optional
creatable functionality to support cases where SpEL is allowable.

### CODE REVIEW

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  private getNamespaceOptions(): Array<Option<string>> {
+    const { accounts, selectedAccount } = this.props;
+    const selectedAccountDetails = find(accounts, a => a.name === selectedAccount);
+    const namespaces = get(selectedAccountDetails, 'namespaces', []);
+    const options = map(namespaces, n => ({ label: n, value: n }));
+    if (this.props.createable && !includes(namespaces, this.props.selectedNamespace)) {
+      options.push({ label: this.props.selectedNamespace, value: this.props.selectedNamespace });
+    }
+    return options;
```
Comment (ethanfrogers) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx]: This block of code isn't very elegant. I'm sure there's a cleaner way to do this.

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export class NamespaceSelector extends React.Component<INamespaceSelectorProps> {
+  public defaultProps = { createable: false };
+
+  private getNamespaceOptions(): Array<Option<string>> {
+    const { accounts, selectedAccount } = this.props;
+    const selectedAccountDetails = find(accounts, a => a.name === selectedAccount);
+    const namespaces = get(selectedAccountDetails, 'namespaces', []);
+    const options = map(namespaces, n => ({ label: n, value: n }));
+    if (this.props.createable && !includes(namespaces, this.props.selectedNamespace)) {
```
Comment (maggieneterval) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx]: In the situation where a user has selected a namespace available in a particular account, and then changes the account to one where that namespace is no longer an option, we will keep that namespace available as an option, which might be confusing. Wonder if there's any way to designate a particular option as being user-created and only persist those between account re-selects?

Comment (ethanfrogers) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx]: Oh, that's a good catch. I didn't consider that. What if we just limited it to options containing `${`? That should satisfy the use case without having to worry about the above. The option would only persist if you're using SPeL.

Comment (maggieneterval) [reply app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx]: love it, ship it

**Comment by ethanfrogers** [comment]
@maggieneterval updated the check to `this.props.creatable && selectedNamespace.includes('${')`

File: app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  public render() {
+    const componentProps = {
+      clearable: false,
+      options: this.getNamespaceOptions(),
+      value: this.props.selectedNamespace,
+      onChange: (option: Option) => this.props.onChange(option.value.toString()),
+    };
+    return <>{this.props.createable ? <Creatable {...componentProps} /> : <Select {...componentProps} />}</>;
```
Comment (maggieneterval) [inline app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/NamespaceSelector.tsx]:
```suggestion
    return this.props.createable ? <Creatable {...componentProps} /> : <Select {...componentProps} />;
```
I don't think wrapping in a fragment is necessary here

**Pull request merged**